tokio/tests/uds_datagram.rs

```diff
@@ -411,3 +411,47 @@ async fn poll_ready() -> io::Result<()> {
 
     Ok(())
 }
+
+#[tokio::test(flavor = "current_thread")]
+async fn coop_uds() -> io::Result<()> {
+    use std::sync::atomic::{AtomicU64, Ordering};
+    use std::time::{Duration, Instant};
+
+    const HELLO: &[u8] = b"hello world";
+    const DURATION: Duration = Duration::from_secs(1);
+
+    let dir = tempfile::tempdir().unwrap();
+    let server_path = dir.path().join("server.sock");
+
+    let client = std::os::unix::net::UnixDatagram::unbound().unwrap();
+    let server = UnixDatagram::bind(&server_path).unwrap();
+
+    let counter = Arc::new(AtomicU64::new(0));
+
+    let counter_jh = tokio::spawn({
+        let counter = counter.clone();
+
+        async move {
+            loop {
+                tokio::time::sleep(Duration::from_millis(250)).await;
```

Comment:
i would recommend avoiding time in tests like this, and instead using yield_now to ensure that tasks wait a certain number of ticks rather than an amount of time. timing makes things brittle

Reply:
What I'd recommend doing here is rewriting this test such that you have the main task use a pair of UDS sockets to message itself and read the datagram message in a for loop which runs just often enough that it should deplete its budget once, with a second task spawned before the loop which sets a boolean flag and exits to indicate that it run.

You'd then assert after the loop in the main task that the flag was flipped and the second task ran.